server/src/main/java/org/apache/kafka/server/config/KafkaConfig.java

```diff
@@ -0,0 +1,420 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.config;
+
+import org.apache.kafka.common.compress.GzipCompression;
+import org.apache.kafka.common.compress.Lz4Compression;
+import org.apache.kafka.common.compress.ZstdCompression;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.SaslConfigs;
+import org.apache.kafka.common.config.SecurityConfig;
+import org.apache.kafka.common.config.SslClientAuth;
+import org.apache.kafka.common.config.SslConfigs;
+import org.apache.kafka.common.config.TopicConfig;
+import org.apache.kafka.common.config.internals.BrokerSecurityConfigs;
+import org.apache.kafka.common.record.LegacyRecord;
+import org.apache.kafka.common.record.Records;
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+import org.apache.kafka.common.utils.Utils;
+import org.apache.kafka.coordinator.group.ConsumerGroupMigrationPolicy;
+import org.apache.kafka.coordinator.group.Group;
+import org.apache.kafka.coordinator.group.GroupCoordinatorConfig;
+import org.apache.kafka.coordinator.transaction.TransactionLogConfigs;
+import org.apache.kafka.coordinator.transaction.TransactionStateManagerConfigs;
+import org.apache.kafka.network.SocketServerConfigs;
+import org.apache.kafka.raft.QuorumConfig;
+import org.apache.kafka.security.PasswordEncoderConfigs;
+import org.apache.kafka.server.common.MetadataVersionValidator;
+import org.apache.kafka.server.metrics.MetricConfigs;
+import org.apache.kafka.server.record.BrokerCompressionType;
+import org.apache.kafka.storage.internals.log.CleanerConfig;
+import org.apache.kafka.storage.internals.log.LogConfig;
+
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+
+import static org.apache.kafka.common.config.ConfigDef.Importance.HIGH;
+import static org.apache.kafka.common.config.ConfigDef.Importance.LOW;
+import static org.apache.kafka.common.config.ConfigDef.Importance.MEDIUM;
+import static org.apache.kafka.common.config.ConfigDef.Range.atLeast;
+import static org.apache.kafka.common.config.ConfigDef.Range.between;
+import static org.apache.kafka.common.config.ConfigDef.Type.BOOLEAN;
+import static org.apache.kafka.common.config.ConfigDef.Type.CLASS;
+import static org.apache.kafka.common.config.ConfigDef.Type.DOUBLE;
+import static org.apache.kafka.common.config.ConfigDef.Type.INT;
+import static org.apache.kafka.common.config.ConfigDef.Type.LIST;
+import static org.apache.kafka.common.config.ConfigDef.Type.LONG;
+import static org.apache.kafka.common.config.ConfigDef.Type.PASSWORD;
+import static org.apache.kafka.common.config.ConfigDef.Type.SHORT;
+import static org.apache.kafka.common.config.ConfigDef.Type.STRING;
+
+public class KafkaConfig {
```

Comment:
If it is used to return `ConfigDef` only, maybe it can be renamed to `KafkaConfigDefs`?

Reply:
I was planning to move the rest of KafkaConfig to it and make this class an abstract in the next PR and KafkaConfig.scala will extend it for now until I move everything out

Reply:
> I was planning to move the rest of KafkaConfig to it and make this class an abstract in the next PR and KafkaConfig.scala will extend it for now until I move everything out

not sure whether I catch your point. it seems we will have a java version KafkaConfig but it is smaller?

Reply:
I was planning to keep 2 KafkaConfig for the time being while moving everything out of core. 
Using `KafkaConfigDef` has the following 
**Pros:** 
-  The ConfigDef is huge so having this out of `KafkaConfig` whether it is the original `KafkaConfig` in `core` or the new `KafkaConfig` in `server`. 
**Cons:** 
- We will have class that only contains 1 static variable and most likely not going to scale for any thing else. 

Having 2 `KafkaConfig` (most likely the one in server will keep getting smaller and smaller till we are ready to delete). The idea I hade that the one in `server` will be an abstract class that will be extended by the one in `core` and I slowly will move stuff out of `core` to `server` `KafkaConfig`
**Pros:** 
- easier to review as all references and imports will point to KafkaConfig in core during the process of moving the getters into `server`. And the last PR will be just changing the import to the right `KafkaConfig`
- Removing ZK code from `core::KafkaConfig` (which will happened in the next few weeks) will not create conflicts in `server::KafkaConfig` as I can ignore moving ZK `getters` and validation. 
**Cons:**
- We have two `KafkaConfig` for a period of time


I don't mind renaming this to `KafkaConfigDef` but my point is going forward  we most likely going to have 2 `KafkaConfig` for a short period of time.

Reply:
It seems option2 is better. My concern was that `KafkaConfig` will be still mixed with all configs even though we move it to server module.

Reply:
BTW, please fix the conflicts again :)

Reply:
> It seems option2 is better. My concern was that KafkaConfig will be still mixed with all configs even though we move it to server module.

Unfornatly yes. I was planning to make `server::KafkaConfig` as abstract for the time being until I move everything out.

Reply:
> BTW, please fix the conflicts again :)

Fixed now

Reply:
>  I was planning to make server::KafkaConfig as abstract for the time being until I move everything out.

That is a intermediate state, right?  I assume there will be specific config calss for different component (such as `RemoteLogManagerConfig` and `QuorumConfig`). server::KafkaConfig will be either nonexistent or small class (maybe do check for all configs or run once in starting)

Reply:
This would be another piece of work. Right now in the context of [KAFKA-15853](https://github.com/apache/kafka/pull/16116#top) I want to move `KafkaConfig` to unblock the rest of [KAFKA-15852](https://issues.apache.org/jira/browse/KAFKA-15852) that depends on `KafkaConfig`